base/docs/bootstrap.jl

```diff
@@ -15,13 +23,26 @@ macro doc(args...)
 end
 
 setexpand!() do str, obj
-    push!(docs, (current_module(), str, obj))
+    docs.head = Node((ccall(:jl_get_current_module, Any, ()), str, obj), docs.head)
     return esc(obj)
 end
 
+"""
+    DocBootstrap :: Module
+
+Basic docstring bootstrapping module that accumulates docstrings prior to the real docsystem
+being defined in `base/docs/Docs.jl`. Once the proper docsystem is loaded all docstrings
+that were stored in `DocBootstrap.docs` are migrated to their correct modules using
+`DocBootstrap.loaddocs()`.
+"""
+DocBootstrap
+
 function loaddocs()
-    for (mod, str, obj) in docs
+    node = docs.head
+    while node ≠ nothing
+        mod, str, obj = node.data
         eval(mod, :(Base.@doc($str, $obj, false)))
+        node = node.next
     end
```

Comment:
The ``sys.so`` size doesn't change noticeably. 25.9mb with or without. That's with 36 docstrings getting kept in ``docs``. I think I'll still empty it anyway since they aren't really of much use afterwards.